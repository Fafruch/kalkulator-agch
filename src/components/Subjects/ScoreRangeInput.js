import React from 'react'
import PropTypes from 'prop-types'

const ScoreRangeInput = ({ subject: { id, elementaryScore, advancedScore }, examType, subjectsType, onChange }) => {
  const isElementary = examType === 'elementary'

  const handleChange = (event) => {
    const inputScore = +event.target.value

    if (isElementary) elementaryScore = inputScore
    else advancedScore = inputScore

    onChange(id, elementaryScore, advancedScore, subjectsType)
  }

  return (
    <input
      type='range'
      value={isElementary ? elementaryScore : advancedScore}
      min='0'
      max='100'
      onChange={handleChange}
      className='subject-input-range'
    />
  )
}
ScoreRangeInput.propTypes = {
  subject: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    elementaryScore: PropTypes.number.isRequired,
    advancedScore: PropTypes.number.isRequired,
    max: PropTypes.object.isRequired,
    active: PropTypes.bool.isRequired
  }).isRequired,
  examType: PropTypes.string.isRequired,
  subjectsType: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

export default ScoreRangeInput
