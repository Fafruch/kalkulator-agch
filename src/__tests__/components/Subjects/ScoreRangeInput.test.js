import React from 'react'
import { shallow } from 'enzyme'

import ScoreRangeInput from '../../../components/Subjects/ScoreRangeInput'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subject: {
      id: '0',
      elementaryScore: 100,
      advancedScore: 82,
      active: true
    },
    examType: 'elementary',
    subjectsType: 'primary',
    onChange: jest.fn()
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<ScoreRangeInput {...props} />)

  return {
    props,
    wrapper
  }
}

describe('ScoreRangeInput component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should handle score change', () => {
    let { props, wrapper } = setup()

    wrapper.find('input').simulate('change', { target: { value: '10' } })
    expect(props.onChange).toBeCalledWith({ id: '0', elementaryScore: 10, advancedScore: 82, subjectsType: 'primary' })

    ;({ props, wrapper } = setup({ examType: 'advanced' }))

    wrapper.find('input').simulate('change', { target: { value: '8' } })
    expect(props.onChange).toBeCalledWith({ id: '0', elementaryScore: 100, advancedScore: 8, subjectsType: 'primary' })
  })
})
