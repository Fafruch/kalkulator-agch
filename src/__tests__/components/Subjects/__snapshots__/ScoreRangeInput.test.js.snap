// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreRangeInput component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScoreRangeInput
    examType="elementary"
    onChange={[Function]}
    subject={
        Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
          }
    }
    subjectsType="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "className": "subject-input-range",
      "max": "100",
      "min": "0",
      "onChange": [Function],
      "type": "range",
      "value": 100,
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": "subject-input-range",
        "max": "100",
        "min": "0",
        "onChange": [Function],
        "type": "range",
        "value": 100,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
